### PR TITLE
Opt out moving component templates used as partials or in layoutName

### LINKED ISSUE #8: Codemod breaks apps which use `layoutName` and partials
So while both deprecated, if you do either of the following in your application and run this codemod it will break your app.

1. Set component templates using `layoutName`
```js
export default Component.extend({
  layoutName: 'components/attributes/attribute',
});
```
2. Use templates within `app/templates/components` as templates
```hbs
{{partial "components/modal/message"}}
```

Both these mechanisms rely on the templates still appearing within `app/templates/components`. Currently we completely delete the `app/templates/components` directory. Should we instead be more selective and only move those templates which aren't used in `layoutName` or `partial`?

While testing out co-location in our app I found it was annoying to have to manually copy (partial and `layoutName`) template files back to their original locations so the app would work properly. So I went ahead forked this codemod and implemented the behavior where partials and layoutName templates are not moved and left where they are. If you think it's useful, I could submit a PR here with my changes?

### PR DESCRIPTION
**Pull request by patocallaghan**
Fixes #8 

This change allows the codemod to work on apps which use legacy template paradigms such as `layoutName` and `partials`. If a component template is used as either a `layoutName` or a `partial` it's not moved to the co-located template location and left in the `app/templates/components` location instead.

### CODE REVIEW

File: lib/utils/js-parser.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+'use strict';
```
Comment (rwjblue) [inline lib/utils/js-parser.js:1]: FWIW, I would like for us to come up with a more specific mechanism for building up the parser within an Ember app.

Definitely doesn't change anything for now, but food for thought...

File: lib/utils/templates.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  console.info(`Checking if any component templates are used as templates of other components using \`layoutName\``);
+  let names = files.map(file => {
+    let content = readFileSync(file, 'utf8');
+    return fileInLayoutName(content);
+  }).filter(Boolean);
+  return Array.from(new Set(names));
+}
+
+function fileInLayoutName(content) {
+  debugger;
```
Comment (rwjblue) [inline lib/utils/templates.js]: Probably want to kill this line

Comment (patocallaghan) [reply lib/utils/templates.js]: doh! I've  🔪 that

File: lib/utils/js-parser.js (hunk 1 above)
Comment (patocallaghan) [reply lib/utils/js-parser.js:1]: Sure. What exactly were you thinking? Something that builds the parser dynamically? or just moving this sort of logic out to an addon?

Comment (rwjblue) [reply lib/utils/js-parser.js:1]: Yeah, something like that. TBH, I'm not sure, but it is _vaguely_ annoying to have to come up with this for every codemod separately.

Definitely something to think on...

**Review by rwjblue — APPROVED** [review]

File: lib/utils/js-parser.js (hunk 1 above)
Comment (patocallaghan) [reply lib/utils/js-parser.js:1]: I'd be happy to spike something and ping you on it

**Pull request merged**